packages/block-library/src/navigation-link/edit.js

```diff
@@ -544,7 +547,7 @@ export default function NavigationLinkEdit( {
 		if ( hasChildren ) {
 			transformToSubmenu();
 		}
-	}, [] );
+	} );
```

Comment:
Looks like the dependency list was removed?

This code wasn't great in `trunk`, so there's an opportunity to improve it. 

I think it'd be better to split it into two effects. A first for the `setIsLinkOpen` part, which would have `url` as a dep, the other for the `transformToSubmenu` part which would have `hasChildren` as a dep.